### PR TITLE
Update ORM demos

### PR DESCRIPTION
**Pull request by Naros**
Migrated ORM demos to 5.2.13.Final and upgraded the gradle build to use version 4.4 rather than 1.9.

### CODE REVIEW

**Comment by Naros** [comment]
@gsmet Do these address the concerns which you mentioned in PR https://github.com/hibernate/hibernate-demos/pull/29?

File: hibernate-orm/core/Basic/src/main/resources/META-INF/persistence.xml [hunk 1]
```diff
@@ -2,7 +2,17 @@
...
+		<!-- Define the classes -->
+		<class>org.hibernate.brmeyer.demo.entity.User</class>
+		<class>org.hibernate.brmeyer.demo.entity.Tool</class>
+		<class>org.hibernate.brmeyer.demo.entity.Skill</class>
+		<class>org.hibernate.brmeyer.demo.entity.Community</class>
+		<class>org.hibernate.brmeyer.demo.entity.Donation</class>
+		<class>org.hibernate.brmeyer.demo.entity.Comment</class>
+		<class>org.hibernate.brmeyer.demo.entity.ServiceEvent</class>
+		<class>org.hibernate.brmeyer.demo.entity.Project</class>
+
```
Comment (gsmet) [inline hibernate-orm/core/Basic/src/main/resources/META-INF/persistence.xml]: @Naros what's the reason for this change?

Comment (Naros) [reply hibernate-orm/core/Basic/src/main/resources/META-INF/persistence.xml]: @gsmet I specifically noticed that by upgrading to ORM, the `BasicJpaDemo` failed because it couldn't locate the entity classes.  Obviously there are several ways I could inject the annotated classes into the persistence unit configuration, this being just one of them.

Another solution would be to pass a list of annotated classes to the construction of the factory using `AvailableSettings#LOADED_CLASSES` much like how our test suite does.

Do you have a preferred approach?

Comment (gsmet) [reply hibernate-orm/core/Basic/src/main/resources/META-INF/persistence.xml]: I don't have any preference :). I was just surprised they were automatically discovered before and that we need to declare them now.

**Comment by gsmet** [comment]
Nice to have these working again. Will merge this tomorrow morning.

**Comment by gsmet** [comment]
Merged, thanks Chris!

**Comment by Naros** [comment]
No problem, glad to help.
